src/main.rs

```diff
@@ -20,49 +22,62 @@ fn configure_and_try_main() -> Result<()> {
 }
 
 fn try_main(config: Config) -> Result<()> {
-    let pid_file = config.pid_file.to_owned();
-    if let Err(err) = lock(&pid_file) {
-        return Err(anyhow!(
-            "Failed to create lock file {:?}: {} (tip: use --pid-file to \
-            change the location of the lock file) ",
-            &pid_file,
-            err
-        ));
-    }
+    let mut lock_file = lock(&config)?;
+    write_pid_file(&config.pid_file)?;
 
-    let process_res = process(config);
-    if let Err(e) = unlock(&pid_file) {
-        eprint!("Failed to remove pid file {:?}: {}", pid_file, e);
-    }
+    let process_res = process(&config);
+
+    lock_file.unlock().with_context(|| "Could not remove lockfile")?;
+    remove_pid_file(&config.pid_file)?;
```

Comment:
Shouldn't the unlock sequence be the inverse of the lock sequence, i.e. lock -> write pid file and then remove pid file -> unlock? Otherwise we unlock then remove the PID file potentially just updated by another krill-sync process (as if I look at the code in `write_buf()` it says it invokes `File::create()` which is documented as truncating the file if it already exists, i.e it won't fail if it already exists. ?

Reply:
removed pid file